app/controllers/weeks_controller.rb

```diff
@@ -1,6 +1,8 @@
 class WeeksController < ApplicationController
   protect_from_forgery with: :exception
 
+  before_action :authenticate_user!
```

Comment:
This should go in something like `SecureController` that this controller inherits from.